### PR TITLE
Upgrade to v3.0.0 of the AWS Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #1055
Fixes: #1022

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,14 @@ CHANGELOG
...
 ## HEAD (Unreleased)
-_(none)_
+* Upgrade to v3.0.0 of the AWS Terraform Provider
+  **Please Note:**
+  * A number of `deprecated` properties on resources have been removed. You can check the full list in the 
+  [upstream CHANGELOG](https://github.com/pulumi/terraform-provider-aws/blob/upstream-v3.0.0/CHANGELOG.md#300-july-31-2020)
+  * v3.0.0 introduces:
+    * Improved Credential Ordering
+    * Major changes to the `aws.acm.Certificate` to fix a number of issues regarding perpetual diffs of propertes
+    * Enabling drift detection of `loadBalancers` and `targetGroupArns` in `aws.autoscaling.Group`
```
Comment (lukehoban) [inline CHANGELOG.md:12]: Can we structure this to make more clear that the full set of changes are in the upstream CHANGELOG - but that some of the most notable changes are listed explicitly here?  Currently it reads like there are some deprecated properties removed and then three other changes - which I don't think is true?

File: sdk/nodejs/appmesh/virtualRouter.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ import * as utilities from "../utilities";
  * import * as aws from "@pulumi/aws";
  *
  * const serviceb = new aws.appmesh.VirtualRouter("serviceb", {
- *     meshName: aws_appmesh_mesh_simple.id,
+ *     meshName: aws_appmesh_mesh.simple.id,
```
Comment (lukehoban) [inline sdk/nodejs/appmesh/virtualRouter.ts:31]: I know this isn't related to this PR - but is this "right" in terms of codegen?  This is a reference to an non-existant variable - so there's sort of no real notion of "right" - but the previous code looks more like what I would expect for a reference to an undefined variable.

cc @pgavlin @lblackstone

File: sdk/nodejs/appsync/resolver.ts [hunk 3]
```diff
@@ -70,23 +64,29 @@ import * as utilities from "../utilities";
  *     $utils.appendError($ctx.result.body, $ctx.result.statusCode)
  * #end
  * `,
- *     type: "Query",
+ *     cachingConfig: {
+ *         cachingKeys: [
+ *             `$context.identity.sub`,
```
Comment (lukehoban) [inline sdk/nodejs/appsync/resolver.ts:69]: Is it intentional that we changed lots of string literals to use backticks?  Are we doing this when there is a literla `$` in the string now?  

cc @pgavlin @lblackstone

File: sdk/nodejs/appmesh/virtualRouter.ts (hunk 2 above)
Comment (pgavlin) [reply sdk/nodejs/appmesh/virtualRouter.ts:31]: I believe that this was caused by the change from 0.11 to 0.12 syntax: https://github.com/terraform-providers/terraform-provider-aws/commit/d47e82f170c20e655a30557289ef3772c44e13a8#diff-0601a42442cec4be2db943fba3998311

With TF12, we render this into a traversal without trying to infer the kind of the missing reference. We could improve that and retain the original rendering by doing that inference in `tf2pulumi`.

File: sdk/nodejs/appsync/resolver.ts (hunk 3 above)
Comment (pgavlin) [reply sdk/nodejs/appsync/resolver.ts:69]: Same answer re: the transition to TF12 syntax. I'm not 100% sure what's triggering this, but the `$`s are a good guess.

**Comment by lukehoban** [comment]
> Is it possible to seperate those two things? It's hard to tell what the v3.0.0 changes are as there is a lot of noise (and a number of questions on whether the codegen changes are even "right") mixed in.

From @pgavlin’s other replies it sounds sa like the answer is “no” since these changes are coming from examples moving from TF0.11 to TF0.12. So will just need to review these together.

**Comment by stack72** [comment]
Merged as 6c13f319a64f3fdf2c71cb3808715a12c2322f0d

**Pull request closed without merging**